dist/src/main/java/io/camunda/zeebe/gateway/GatewayClusterConfiguration.java

```diff
@@ -87,7 +87,9 @@ private MessagingConfig messagingConfig(final GatewayCfg config) {
         new MessagingConfig()
             .setCompressionAlgorithm(cluster.getMessageCompression())
             .setInterfaces(Collections.singletonList(cluster.getHost()))
-            .setPort(cluster.getPort());
+            .setPort(cluster.getPort())
+            // TODO: mark as experimental, make it configurable
+            .setMessagingMetrics(true);
```

Comment:
:x: I dislike this.

@npepinpe I need your input here. Do you think it makes sense to add for the gateway also a ExperimentConfig, I feel yes, but this might be a bigger thing. Any opinion on this one?

Reply:
Are the metrics expensive? :thinking: If not, we can leave them on all the time (like most metrics).

If so, then yeah it makes sense to add an experimental config for the gateway (which would be useful long term anyway).

Reply:
I was not 100% sure, since we have as label right now the receiver (IP). But this should change that often idk.

Reply:
Yeah an IP is not a good label because it will have high cardinality. Can we use anything else? The best would be member ID, but I guess the messaging service doesn't know that. Can't we use the host/DNS instead of the IP? Though users which configure using IPs would still have high cardinality (though I don't think that's a common prod deployment)

Reply:
There was a reason for using IP, since we had before the problem that the IP was not correctly reset/resolved again which I would like to have it here to see to which endpoint the request is sent.

Reply:
But as a label it has high cardinality. Can we maybe find a different way to spot the problem? :thinking:

Reply:
> BTW this shouldn't block us, so please keep going with IPs for now if we don't have a good idea

Means enabling per default? Or make it configurable in the gateway as well?

Reply:
I would recommend enabling by default + follow up issue to address the high cardinality that must be done before we release in April (where we either figure out how to not use IPs, or we put behind a feature flag). WDYT?

Reply:
I think Meggle wanted to tackle organizing issues per releases so we can plan them when its convenient :thinking: